internal/result/results.go

```diff
@@ -29,13 +30,54 @@ const (
 	FailRunning = false
 )
 
+var (
+	statusSeverity = map[statepb.Row_Result]int{
+		statepb.Row_NO_RESULT:         0,
+		statepb.Row_PASS:              1,
+		statepb.Row_BUILD_PASSED:      2,
+		statepb.Row_PASS_WITH_ERRORS:  3,
+		statepb.Row_PASS_WITH_SKIPS:   4,
+		statepb.Row_RUNNING:           5,
+		statepb.Row_CATEGORIZED_ABORT: 6,
+		statepb.Row_UNKNOWN:           7,
+		statepb.Row_CANCEL:            8,
+		statepb.Row_BLOCKED:           9,
+		statepb.Row_TOOL_FAIL:         10,
+		statepb.Row_TIMED_OUT:         11,
+		statepb.Row_CATEGORIZED_FAIL:  12,
+		statepb.Row_BUILD_FAIL:        13,
+		statepb.Row_FAIL:              14,
+		statepb.Row_FLAKY:             15,
+	}
+)
+
+func resultLte(rowResult, compareTo state.Row_Result) bool {
+	return statusSeverity[rowResult] <= statusSeverity[compareTo]
+}
+
+func resultGte(rowResult, compareTo state.Row_Result) bool {
+	return statusSeverity[rowResult] >= statusSeverity[compareTo]
+}
+
+// IsPassingResult returns true if the Row_Result is any of the passing results,
+// including PASS_WITH_SKIPS, BUILD_PASSED, and more.
+func IsPassingResult(rowResult state.Row_Result) bool {
+	return resultGte(rowResult, statepb.Row_PASS) && resultLte(rowResult, statepb.Row_PASS_WITH_SKIPS)
+}
+
+// IsFailingResult returns true if the Row_Result is any of the failing results,
+// including CATEGORIZED_FAILURE, BUILD_FAIL, and more.
+func IsFailingResult(rowResult state.Row_Result) bool {
+	return resultGte(rowResult, statepb.Row_TOOL_FAIL) && resultLte(rowResult, statepb.Row_FAIL)
+}
+
 // Coalesce reduces the result to PASS, NO_RESULT, FAIL or FLAKY.
 func Coalesce(result state.Row_Result, ignoreRunning bool) state.Row_Result {
 	// TODO(fejta): other result types, not used by k8s testgrid
 	if result == state.Row_NO_RESULT || result == state.Row_RUNNING && ignoreRunning {
 		return state.Row_NO_RESULT
 	}
-	if result == state.Row_FAIL || result == state.Row_RUNNING {
+	if IsFailingResult(result) || result == state.Row_RUNNING {
```

Comment:
@michelle192837 this deserves unit testing. 

Looks like CANCEL coalesces to passed?

Reply:
https://github.com/GoogleCloudPlatform/testgrid/pull/266